include/triton/core/tritonbackend.h

```diff
@@ -580,6 +603,39 @@ TRITONBACKEND_DECLSPEC TRITONSERVER_Error* TRITONBACKEND_ResponseOutput(
     const char* name, const TRITONSERVER_DataType datatype,
     const int64_t* shape, const uint32_t dims_count);
 
+/// Create an output state tensor. The returned state object is only valid
+/// before the TRITONBACKEND_RequestStateUpdate is called. The state should not
+/// be freed by the caller.
+///
+/// \param response The request.
+/// \param state Returns the new response output state.
+/// \param name The name of the output tensor.
+/// \param datatype The datatype of the output tensor.
+/// \param shape The shape of the output tensor.
+/// \param dims_count The number of dimensions in the output tensor
+/// shape.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONBACKEND_DECLSPEC TRITONSERVER_Error* TRITONBACKEND_RequestStateNew(
```

Comment:
Think of each state as similar to a response. So have TRITONBACKEND_StateNew, TRITONBACKEND_StateDelete and TRITONBACKEND_StateUpdate. Be sure to document lifecycle and ownership of State objects.

Make a TRITONBACKEND_State section for all the state related functions